### DISCUSSION TITLE
Importing from Nordigen duplicates and matches

### DISCUSSION THREAD

**Discussion by AndreiGavriliu** [firefly-iii/firefly-iii#6017] [Firefly III]
Hi all,

I'm running FF on kubernetes and using a cronjob to import the last 2 days from Nordigen every 6 hours (just to be sure I get everything imported if something goes wrong). After a few weeks I noticed the following issues:

Duplicates:
* the External ID from Nordigen changes from one day to the other, thus the transaction gets imported twice
* I tried the classic approach, but this also fails because the description changes. On the first day, the description contains a lot of blank spaces, on the second day, the spaces are gone, and it looks "correct", but, since they don't match, the transaction gets imported again

Destination account:
* I do not seem to be able to map the destination account. I'd like to have the Opposing IBAN added to "destination account"

Any Ideas?

Thanks,
Andrei

**Comment by JC5** [comment]
> (just to be sure I get everything imported if something goes wrong)

Haha I do the same.

> On the first day, the description contains a lot of blank spaces, on the second day, the spaces are gone

If you could send me some examples, I may be able to raise this with Nordigen. Although this is probably something your bank has to fix.

> I do not seem to be able to map the destination account.

The opposing IBAN should be added to the transaction array that's submitted to Firefly III.

**Comment by AndreiGavriliu** [comment]
I finally got around to contact the bank and, apparently, after the transactions are created initially, they modify them later on and that's why Nordigen thinks they are new. The description changes, they are categorized, source/dest accounts are also "corrected". They suggested I import only "yesterday", since "today" will be updated over night.

I checked the docs but could not find anything about passing the date-range via POST. Did I miss anything or is it just not possible?

**Reply by JC5** [reply]
It's not possible. At least not with a end date that's not today. 

https://github.com/firefly-iii/firefly-iii/issues/6061

**Comment by joseluislucio** [comment]
> using a cronjob to import the last 2 days from Nordigen every 6 hours

How do you do that? could you please paste an example?

**Reply by JC5** [reply]
https://docs.firefly-iii.org/data-importer/usage/command_line/

Combine it with a cron job 

https://www.google.com/search?client=firefox-b-d&q=set+up++a+cron+job

**Comment by joseluislucio** [comment]
I don't know what am I doing wrong ... but here is what I have done:

As root on my raspberry Pi I have created the following con job with contab -e

![image](https://user-images.githubusercontent.com/1066785/186128516-2b3e75ee-c89f-49e7-877b-bd4fc30a9b5f.png)

If I check the logs with  grep CRON /var/log/syslog ... i get this:

Aug 23 11:30:01 raspberrypi CRON[11618]: (root) CMD (docker exec -it Firefly3_FIDI php artisan importer:import /var/www/html/storage/upload/configurations/Caixabank2D.json)

So ... the cron job is launched as expected

If I check mail ... I see this:
![image](https://user-images.githubusercontent.com/1066785/186129223-2816ab15-163e-4c4b-bd58-e06d00276ff7.png)
![image](https://user-images.githubusercontent.com/1066785/186129320-2dccc162-1dd1-4eec-a992-b21687043c6d.png)

Looks like it is working ... but nothing updated in the app.

**If I lauch the command** (docker exec -it Firefly3_FIDI php artisan importer:import /var/www/html/storage/upload/configurations/Caixabank2D.json) **manualy ... it works like a charm.**

Any tip?

Regards

**Reply by JC5** [reply]
You're running Docker exec with `-it`. Check out what those mean and you'll see why it doesn't work 😉

**Reply by joseluislucio** [reply]
> You're running Docker exec with `-it`. Check out what those mean and you'll see why it doesn't work 😉

Remove -it ... and looks like it is working

**Comment by joseluislucio** [comment]
I can confirm that it's working. Thanks a lot for your help.
